src/main/java/org/opentripplanner/graph_builder/module/osm/WayPropertySetSource.java

```diff
@@ -13,16 +13,50 @@
 
 package org.opentripplanner.graph_builder.module.osm;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.MissingNode;
+
 /**
  * Factory interface for providing a {@link WayPropertySet} that determine how OSM
  * streets can be traversed in various modes and named.
  *
- * TODO clarify why we need a factory interface for this.
- *
- * @author bdferris, novalis
- * @see WayProperty
- * @see OpenStreetMapModule#setWayPropertySetSource(WayPropertySetSource)
+ * @author bdferris, novalis, seime
  */
 public interface WayPropertySetSource {
+
     public WayPropertySet getWayPropertySet();
+    
+    public class WayPropertySetSourceFactory {
```

Comment:
Can we just eliminate this inner class and put the static method on the containing class?